corehq/apps/translations/app_translations/upload_form.py

```diff
@@ -142,13 +143,14 @@ def _add_missing_translation_elements_to_itext(self, template_translation_el):
             if not trans_el.exists():
                 new_trans_el = copy.deepcopy(template_translation_el.xml)
                 new_trans_el.set('lang', lang)
-                if lang != self.app.langs[0]:
-                    # If the language isn't the default language
-                    new_trans_el.attrib.pop('default', None)
-                else:
-                    new_trans_el.set('default', '')
                 self.itext.xml.append(new_trans_el)
 
+    def _modify_default_lang_if_needed(self, lang, trans_el):
```

Comment:
Maybe it's weird to bring up now since we paired on this, but I don't love the name, especially since we're always adding or removing the `default` attribute whether a change is needed or not. 

What would you think about something in the realm of `_set_translation_node_default_lang` (wordy but gets to the core of what's happening) or `_update_node_default_attr` (concise and more literally what's happening)?

Reply:
Hah you're allowed to voice opinions after we paired. I agree that we indiscriminately attempt to update the node, but that is an implementation detail of the method that I don't think _has_ to be reflected in the method name. The intent is still that if the node already has the 'default' attr set, nothing changes (and vice versa).

So I guess the question you're getting at is do we get value out of appending `if_needed` or `if_necessary` to this method name, or would it only cause confusion once a dev looks at the implementation which I could imagine could be true. To take one out of Matt's playbook, if we try to say in plain english what the intent of this method is, what do we get? I would say we are making sure that the current node is up to date with the app's default language. So maybe something like `def _update_default_language(self, node)`? I'm in favor of changing the method to look up the language on the node as well to simplify things.

Reply:
I'm not opposed to `if_needed`, I think I'm just wanting it to be _really clear_ where this is updating the default and based on what criteria, since the multiple places we keep a "default" seem to be part of what made investigating more difficult. I think argument naming might actually help a lot here. With all this in mind I think my preference would be something close to `def _set_language_as_default_if_needed(self, node_lang, node)`

Reply:
Yeah realistically I think a comment is worthwhile since this is a bit of a workaround/hack anyway, and we won't be able to capture all of that context in just a method name. I know this contradicts my statement about abstracting implementations away, but I think I'm content with `_update_default_attr_if_needed(node, node_lang)`.

c35a5dc31de08757b8e7e0391834f560da03b9bb

Reply:
Ok actually I think most of our issues were because where we were calling this method, not the method name. I changed this in 1c503f4881f85ac05f0a675ed47ae670841ce477 so that it is hopefully more intuitive what is happening and why.